ADAL/src/cache/ios/ADKeychainTokenCache.m

```diff
@@ -451,6 +451,34 @@ - (BOOL)removeAllForUserId:(NSString * __nonnull)userId
     return deleteSuccessful;
 }
 
+- (BOOL)removeAllForUserId:(NSString *)userId error:(ADAuthenticationError *__autoreleasing  _Nullable *)error
+{
+    AD_LOG_WARN_DICT(([NSString stringWithFormat:@"Removing all items for user <%@>", userId]), nil,
+                     (@{ @"operation" : @"removeAllForClientId:", @"userId" : userId }), nil);
+
+    BOOL deleteSuccessful = YES;
+    NSArray *items = [self allItems:nil];
+
+    ADAuthenticationError *adError = nil;
+
+    for (ADTokenCacheItem *item in items)
+    {
+        if ([userId isEqualToString:[[item userInformation] userId]])
+        {
+            [self removeItem:item error:&adError];
+            if (adError)
```

Comment:
should check bool return value, not adError

Reply:
Refactoring, no need for adError and the deleteSuccessful boolean.
Previously, it was going through all items even though there was an error along the way.